src/main/java/ru/testtask/service/FileService.java

```diff
@@ -0,0 +1,250 @@
+package ru.testtask.service;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBObject;
+import com.mongodb.MongoWriteException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.index.CompoundIndexDefinition;
+import org.springframework.data.mongodb.core.index.Index;
+import org.springframework.data.mongodb.core.index.TextIndexDefinition;
+import org.springframework.stereotype.Service;
+import ru.testtask.dto.UploadFileDTO;
+import ru.testtask.exception.FileIsToLargeException;
+import ru.testtask.exception.NameAlreadyExistsException;
+import ru.testtask.model.FileData;
+import ru.testtask.repo.FileDataRepo;
+
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+public class FileService {
+
+    @Value("${storage.root}")
+    private String storageRoot;
+
+    private Path storageRootPath;
+
+    private Path temporaryDir;
+
+    @Value("${storage.file.max-size.mb}")
+    private long fileMaxSizeMb;
+
+    private final FileDataRepo fileDataRepo;
+
+    public FileService(FileDataRepo fileDataRepo, MongoTemplate mongoTemplate) {
+        this.fileDataRepo = fileDataRepo;
+    }
+
+    @PostConstruct
+    public void init(){
+        storageRootPath = Paths.get(storageRoot);
+        if (Files.notExists(storageRootPath)){
+            try {
+                Files.createDirectories(storageRootPath);
+            } catch (IOException e) {
+                log.error(String.format("It is impossible to create storage root directory : %s", storageRoot), e);
+                System.exit(-1);
+            }
+        }
+    }
+
+    public FileData uploadFile(UploadFileDTO uploadFileDTO) throws FileIsToLargeException{
+        try {
+            URL url = new URL(uploadFileDTO.getFileUrl());
+            String filename = UUID.randomUUID().toString() + "." + FilenameUtils.getExtension(url.toString());
+            temporaryDir = Files.createTempDirectory(storageRootPath, "temp");
```

Comment:
You change the variable each time when upload is starting. The variable is used from all threads that execute uploading. What happens if you will start multiple uploads in parallel?